pilot/pkg/security/authz/policy/helper.go

```diff
@@ -124,15 +125,19 @@ func SimpleRole(name string, namespace string, service string) *model.Config {
 	}
 }
 
-func SimplePrincipal(name string) string {
-	return fmt.Sprintf("cluster.local/ns/%s/sa/%s", name, name)
-}
-
 func CustomPrincipal(trustDomain, namespace, saName string) string {
 	return fmt.Sprintf("%s/ns/%s/sa/%s", trustDomain, namespace, saName)
 }
 
-func SimpleBinding(name string, namespace string, role string) *model.Config {
+func BindingTag(name string) string {
+	return fmt.Sprintf("UserFromBinding[%s]", name)
+}
+
+func SimpleBinding(name, namespace, role, user string) *model.Config {
```

Comment:
Can you not to change this function? you can create a `SimpleBindingWithUser()`, and change `SimpleBinding()` to call it, then you don't need the special check of `""` and also do not need to pass the extra `""` in the tests.

Reply:
Sure